examples/misc/lib/effective_dart/docs_good.dart

```diff
@@ -64,19 +68,17 @@ void miscDeclAnalyzedButNotTested() {
 
   <T>() {
     // #docregion third-person
-    /// Returns `true` if every element satisfies the [predicate].
-    bool all(bool predicate(T element)) => ellipsis();
+    /// Connects to server and fetches results of query.
+    Stream<QueryResult> fetchResults(Query query) => ellipsis();
 
     /// Starts the stopwatch if not already running.
-    void start() {
-      ellipsis();
-    }
+    void start() => ellipsis();
 
     // #enddocregion third-person
   };
 
   // #docregion code-sample
-  /// Returns the lesser of two numbers.
+  /// The lesser of two numbers.
```

Comment:
```suggestion
  /// Gets the lesser of two numbers.
```

I think we should always start with a verb BUT we should avoid using `Returns` since it's self explanatory. I would advise that we use `produces` or `gets` when there isn't a stronger verb.

Reply:
I disagree very strongly.

My goal here is to break away from the unconditional rule that function docs must start with a verb, because it's inconsistent with treating some functions as properties when naming them.

If the function is *named* using a noun phrase or predicate, it's because it is considered a (parameterized) *property*. Then it should be documented the same way it's named.

We also have a naming rule that says "don't start a function name with `get`."
You shouldn't start a DartDoc with "Gets" either, for all the same reasons.

The *reason* for that rule is that it doing so is a sign that the function is really a property.
We then have another rule which says to name such a function it like a property.
That should not just mean *name* it, but to treat it like a property in *every* way, including documentation, and we have rules for documenting properties that don't start with a verb.
Making one a noun and the other a verb makes the documentation and name feel out of sync.

So just "The foo", it's cleaner. The "Gets" adds nothing, but it weakens the connection between name and doc.

Compare:
```dart
/// The first element of this list.
E get first;

/// The [index]th element of this list.
E operator [])(int index);

/// The [index]th element of this iterable.
E elementAt(int index);
```

Adding "Gets" to the `operator[]` function or `elementAt` function doesn't make anything clearer.
It doesn't say anything other than what "Returns" say.

Reply:
I see where you're coming from now. Stepping back from my suggestion.